x/crosschain/keeper/msg_server_update_tss.go

```diff
@@ -70,3 +72,9 @@ func (k msgServer) UpdateTssAddress(
 
 	return &types.MsgUpdateTssAddressResponse{}, nil
 }
+
+// GetChainsSupportingMigration returns the chains that support migration.
+func (k *Keeper) GetChainsSupportingMigration(ctx sdk.Context) []chains.Chain {
```

Comment:
We should add another check:
chain.CCTXGateway == observer

Chains not using the observer solution don't need TSS migration

Reply:
e32c2a87997922c99ad8a9e7709df90d10050b69